state/factory/factory.go

```diff
@@ -98,7 +98,8 @@ type (
 		twoLayerTrie             trie.TwoLayerTrie // global state trie, this is a read only trie
 		dao                      db.KVStore        // the underlying DB for account/contract storage
 		timerFactory             *prometheustimer.TimerFactory
-		workingsets              *lru.Cache // lru cache for workingsets
+		workingsets              *cache.ThreadSafeLruCache            // lru cache for workingsets
+		stateCaches              map[string]*cache.ThreadSafeLruCache // map having lru cache to store current states for speed up
```

Comment:
on an alternative route, think adding this LRU together with the KVStore? something like

```go
type lruKv struct {
    kv db.KVStore
    cache map[string]*cache.ThreadSafeLruCache
}

func AddLRU(kv db.KVStore, cacheSize size) db.KVStore
```
then we can have 
```go
factory.StatedbCacheOption() {
   sf.dao = AddLRU(sf.dao, 1000)
}
```
same thing for statedb.go

Reply:
Modify in this way on the latest commit!